### PR TITLE
CI build updates

### PR DESCRIPTION
**Pull request by alpe**
* Replace Gopath tendermint update with docker
* Build bcpd docker artifact via travis

Note to reviewer: bpcd tests are executed vie main Makefile and bcpd/Makefile. I did not optimize this

### CODE REVIEW

File: examples/mycoind/Makefile [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
-BUILD_FLAGS := -ldflags "-X github.com/iov-one/weave.Version=$(GIT_VERSION)"
-TENDERMINT := ${GOBIN}/tendermint
+BUILD_VERSION ?= $(shell git describe --tags)
+BUILD_FLAGS := -ldflags "-X github.com/iov-one/weave.Version=$(BUILD_VERSION)"
 BUILDOUT ?= mycoind
-GOPATH ?= $$HOME/go
 
-TM_VERSION := v0.21.0
+
+dist: clean test build
```
Comment (ethanfrey) [inline examples/mycoind/Makefile]: this is for consistency , right?

File: cmd/bcpd/Makefile [hunk 2]
```diff
@@ -1,25 +1,18 @@
-.PHONY: all install build test cover deps tools prototools protoc clean dist
+.PHONY: all build test image tf protoc clean dist
```
Comment (ethanfrey) [inline cmd/bcpd/Makefile]: nice an cleaner... why did you pull out tf, btw?
i guess you need to `make tf` in the command dir now, not in root level.

Comment (ethanfrey) [reply cmd/bcpd/Makefile]: The test code need tendermint installed to run properly.

https://github.com/iov-one/weave/blob/merge-bcp-demo/cmd/bcpd/commands/start_test.go#L59-L63

File: examples/mycoind/Makefile (hunk 1 above)
Comment (alpe) [reply examples/mycoind/Makefile]: mainly for the build artifact but as we are not doing any I am going to remove this.

**Comment by ethanfrey** [comment]
Looks good. Thanks

**Pull request merged**